### ISSUE TITLE
Change URL format for package pages

### ISSUE THREAD

**Issue by daveverwer** [SwiftPackageIndex/SwiftPackageIndex-Server#176]
URL format for package pages should be:

`/author/package`

**Comment by finestructure** [comment]
This is sorted out, I just need to clean up the PR. Meanwhile, here are my notes regarding performance and indexes.

id lookup is, of course, blazing fast:

```
Index Scan using "uq:repositories.package_id" on repositories  (cost=0.28..8.30 rows=1 width=242) (actual time=0.021..0.032 rows=1 loops=1)
  Index Cond: (package_id = '2144b337-6e35-4a96-8cd2-8491e8390561'::uuid)
Planning Time: 0.134 ms
Execution Time: 0.077 ms
```

This is the case-insensitive lookup without any additional indexes:

```
Limit  (cost=0.28..130.26 rows=1 width=339) (actual time=1.264..1.302 rows=1 loops=1)
  ->  Nested Loop  (cost=0.28..130.26 rows=1 width=339) (actual time=1.253..1.262 rows=1 loops=1)
        ->  Seq Scan on repositories  (cost=0.00..121.96 rows=1 width=242) (actual time=1.221..1.231 rows=1 loops=1)
              Filter: ((owner ~~* 'finestructure'::text) AND (name ~~* 'gala'::text))
              Rows Removed by Filter: 1850
        ->  Index Scan using packages_pkey on packages  (cost=0.28..8.30 rows=1 width=97) (actual time=0.012..0.022 rows=1 loops=1)
              Index Cond: (id = repositories.package_id)
Planning Time: 0.446 ms
Execution Time: 1.380 ms
```

I don’t think 1.4ms is a problem at all but I had a look if we can use an index anyway (this doesn’t). If we add a `pg_trgm` index, we get to:

```
Limit  (cost=44.28..56.32 rows=1 width=339) (actual time=0.074..0.115 rows=1 loops=1)
  ->  Nested Loop  (cost=44.28..56.32 rows=1 width=339) (actual time=0.064..0.072 rows=1 loops=1)
        ->  Bitmap Heap Scan on repositories  (cost=44.00..48.02 rows=1 width=242) (actual time=0.037..0.045 rows=1 loops=1)
              Recheck Cond: (name ~~* 'gala'::text)
              Filter: (owner ~~* 'finestructure'::text)
              Heap Blocks: exact=1
              ->  Bitmap Index Scan on idx_test1  (cost=0.00..44.00 rows=1 width=0) (actual time=0.022..0.030 rows=1 loops=1)
                    Index Cond: (name ~~* 'gala'::text)
        ->  Index Scan using packages_pkey on packages  (cost=0.28..8.30 rows=1 width=97) (actual time=0.010..0.020 rows=1 loops=1)
              Index Cond: (id = repositories.package_id)
Planning Time: 0.461 ms
Execution Time: 0.262 ms
```

For some reason the index is only used on the `name`. I suspect that’s the planner having insight on the index content. Anyway, I’m going to leave it there and add the following migration:

```
-- prepare
create extension pg_trgm;
create index idx_test1 on repositories using gin (name gin_trgm_ops);
```

```
-- revert
drop index idx_test1;
drop extension pg_trgm;
```

I also need to remove the existing id based lookup. Should have that wrapped up in the morning!

**Comment by daveverwer** [comment]
This looks great 👍

**Comment by finestructure** [comment]
Closed by #253
